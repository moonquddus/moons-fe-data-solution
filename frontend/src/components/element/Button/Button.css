.e-button {
  border: none;
  border-radius: 8px;
  padding: 8px 16px;
  background: var(--calliper-color-navy);
  color: var(--calliper-color-white);
}

.e-button:hover,
.e-button:active,
.e-button:focus {
  background: var(--calliper-color-navy-3);
  cursor: pointer; 
}