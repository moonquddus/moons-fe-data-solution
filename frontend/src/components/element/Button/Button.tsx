import './Button.css'
/**
 * There is plenty more I can do here
 * eg. different themes, sizes, a disabled state, etc
 */
function Button(props: React.PropsWithChildren<React.ButtonHTMLAttributes<HTMLButtonElement>>) {
  const {children, ...data} = props
  const baseClass = 'e-button'
  return (
    <button {...data} className={`${baseClass} ${data.className ?? ''}`}>
      {children}
    </button>
  )
}

export default Button
