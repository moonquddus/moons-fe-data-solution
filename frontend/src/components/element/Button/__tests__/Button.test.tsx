import { render } from '@testing-library/react'
import Button from '../Button'

describe('rendering button', () => {
  it('should render children inside the button', () => {
    const wrapper = render(<Button>Hello I'm a button</Button>)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})
