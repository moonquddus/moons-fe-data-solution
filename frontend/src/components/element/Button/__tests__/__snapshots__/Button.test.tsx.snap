// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering button should render children inside the button 1`] = `
<body>
  <div>
    <button
      class="e-button "
    >
      Hello I'm a button
    </button>
  </div>
</body>
`;
